### PR TITLE
In DAG dependency detector, use class type instead of class name, 2nd attempt

### PR DESCRIPTION
**Pull request by davidavdav**
This is a rebased version of #19683, which got closed by a bot.

### CODE REVIEW

**Comment by potiuk** [comment]
You definietely need to rebase:

```
This branch is 1 commit ahead, 129 commits behind apache:main.
```

File: tests/serialization/test_dag_serialization.py [hunk 1]
```diff
@@ -1295,6 +1295,67 @@ class DummyTask(BaseOperator):
             '<TIDep(Trigger Rule)>',
         ]
 
+    def test_derived_dag_deps_sensor(self):
+        """
+        Tests DAG dependency detection for sensors, including derived classes
+        """
+        from airflow.operators.dummy import DummyOperator
```
Comment (eladkal) [inline tests/serialization/test_dag_serialization.py]: You will need to replace this import with `EmptyOperator`

Comment (davidavdav) [reply tests/serialization/test_dag_serialization.py]: Thanks, I could not interpret the error message in the CI tests correctly

Comment (eladkal) [reply tests/serialization/test_dag_serialization.py]: You will need also to rebase. The error message you had in CI is resolved in https://github.com/apache/airflow/pull/23010

File: tests/serialization/test_dag_serialization.py [hunk 2]
```diff
@@ -1295,6 +1295,67 @@ class DummyTask(BaseOperator):
             '<TIDep(Trigger Rule)>',
         ]
 
+    def test_derived_dag_deps_sensor(self):
+        """
+        Tests DAG dependency detection for sensors, including derived classes
+        """
+        from airflow.operators.dummy import EmptyOperator
```
Comment (eladkal) [inline tests/serialization/test_dag_serialization.py]:
```suggestion
        from airflow.operators.empty import EmptyOperator
```

File: tests/serialization/test_dag_serialization.py [hunk 3]
```diff
@@ -1295,6 +1295,67 @@ class DummyTask(BaseOperator):
...
+                    'dependency_type': 'sensor',
+                    'dependency_id': 'task1',
+                }
+            ]
+
+    def test_derived_dag_deps_operator(self):
+        """
+        Tests DAG dependency detection for operators, including derived classes
+        """
+        from airflow.operators.dummy import EmptyOperator
```
Comment (eladkal) [inline tests/serialization/test_dag_serialization.py]:
```suggestion
        from airflow.operators.empty import EmptyOperator
```

**Review by eladkal — APPROVED** [review]
Looks OK to me.
cc @kaxil @dstandish as you reviewed the previous stale PR

**Review by ashb — APPROVED** [review]

**Pull request merged**